lib/src/base-styles/_nx-btn.scss

```diff
@@ -6,48 +6,64 @@
  */
 @use '../scss-shared/nx-container-helpers';
 @use '../scss-shared/nx-text-helpers';
+@use '../scss-shared/nx-dark-mode-helpers';
 
-%button-focus-outline {
-  outline-offset: -5px;
-  outline: 2px solid;
+@mixin solid-btn-styling($variant) {
+  background-color: var(--nx-color-#{$variant}-button-background);
+  border-color: transparent;
+  color: var(--nx-color-#{$variant}-button-text);
+
+  &:hover {
+    background-color: var(--nx-color-#{$variant}-button-background-hover);
+    color: var(--nx-color-#{$variant}-button-text);
+  }
+
+  &:focus:not(:active) {
+    outline-color: var(--nx-color-#{$variant}-button-focus-outline);
+    // provide 2px of spacing between the edge and the focus outline
+    outline-offset: calc(0px - 2px - var(--nx-focus-outline-width));
+  }
+
+  &:active {
+    background-color: var(--nx-color-#{$variant}-button-background-active);
+    border-color: transparent;
+  }
 }
 
 .nx-btn {
   @include nx-text-helpers.semi-bold;
   @include nx-container-helpers.container-horizontal;
 
-  background-color: var(--nx-swatch-white);
-  border: 1px solid var(--nx-swatch-blue-40);
+  background-color: var(--nx-color-secondary-button-background);
+  border: var(--nx-border-width-button) solid var(--nx-color-secondary-button-border);
   border-radius: var(--nx-border-radius);
-  color: var(--nx-swatch-blue-40);
+  color: var(--nx-color-secondary-button-text);
   cursor: pointer;
   display: inline-block;
   font-size: var(--nx-font-size);
   line-height: var(--nx-line-height);
   margin: 0 var(--nx-spacing-6x);
-  padding: calc(var(--nx-spacing-2x) - 1px) calc(var(--nx-spacing-6x) - 1px);
+  padding: calc(var(--nx-spacing-2x) - var(--nx-border-width-button)) calc(var(--nx-spacing-6x) - var(--nx-border-width-button));
   text-align: center;
   text-decoration: none;
 
   &:hover {
-    background-color: var(--nx-swatch-blue-90);
-  }
-
-  &.nx-btn--secondary:hover {
-    color: var(--nx-swatch-blue-30);
+    background-color: var(--nx-color-secondary-button-background-hover);
+    color: var(--nx-color-secondary-button-text-hover);
   }
 
   &:focus:not(:active) {
-    @extend %button-focus-outline;
-    outline-color: var(--nx-swatch-blue-30);
+    outline: var(--nx-focus-outline-width) solid var(--nx-color-secondary-button-focus-outline);
+    outline-offset: calc(0px - var(--nx-border-width-button) - 4px);
```

Comment:
The `4px` here would also be affected by what I said in https://github.com/sonatype/sonatype-react-shared-components/pull/1038/files#r1158637782